common/views/components/BaseTabs/BaseTabs.tsx

```diff
@@ -0,0 +1,146 @@
+import { useState, useRef, useContext, KeyboardEvent } from 'react';
+import { AppContext } from '@weco/common/views/components/AppContext/AppContext';
+import styled from 'styled-components';
+import { classNames } from '@weco/common/utils/classnames';
+
+const TabList = styled.div.attrs({
+  role: 'tablist',
+})``;
+
+type TabProps = {
+  isActive: boolean;
+  tabPanelId: string;
+};
+
+const Tab = styled.button.attrs((props: TabProps) => ({
+  className: classNames({
+    'plain-button no-padding': true,
+  }),
+  role: 'tab',
+  tabIndex: props.isActive ? 0 : -1,
+  'aria-selected': props.isActive,
+  'aria-controls': props.tabPanelId,
+}))<TabProps>`
+  &:focus {
+    outline: 0;
+  }
+`;
+
+type TabPanelProps = {
+  id: string;
+  isActive: boolean;
+  isEnhanced: boolean;
+};
+const TabPanel = styled.div.attrs((props: TabPanelProps) => ({
+  id: props.id,
+  role: 'tabpanel',
+  hidden: props.isEnhanced && !props.isActive,
+  'aria-expanded': props.isActive,
+}))<TabPanelProps>``;
+
+export type TabType = {
+  id: string;
+  tab: (isActive: boolean, isFocused: boolean) => JSX.Element;
```

Comment:
Is there a reason to have this as a function that returns a component, over a component?
e.g.
```TS
type TabProps = { isActive: boolean, isFocused: boolean }
React.FunctionComponent<TabProps>
````

Reply:
Not that I can think of, I don't think. I can update it.

Reply:
…actually, as a render prop, I think this is a requirement (in order to share the `isActive` and `isFocused` data with the consuming component). Unless I'm missing something.